src/payment/strategies/paypal-commerce/paypal-commerce-payment-processor.ts

```diff
@@ -0,0 +1,91 @@
+import { MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { Overlay } from '../../../common/overlay';
+
+const modalWidth = 450;
+const modalHeight = 600;
+
+export default class PaypalCommercePaymentProcessor {
+    private _window = window;
+    private _popup?: WindowProxy | null;
+    private _overlay: Overlay;
+
+    constructor() {
+        const innerHtml = this._getOverlayElements();
+        this._overlay = new Overlay({}, true, innerHtml);
+    }
+
+    paymentPayPal(approveUrl: string): Promise<boolean> {
+        return new Promise((resolve, reject) => {
+            const paramsWindow =  this._getParamsWindow();
+
+            const closeWindow = (isResolve: boolean, isRemoveOverlay: boolean = true) => {
+                this._window.removeEventListener('message', messageHandler);
+
+                if (this._popup) {
+                    this._popup.close();
+                    this._popup = undefined;
+                }
+
+                if (isRemoveOverlay) {
+                    this._overlay.remove();
+                }
+
+                isResolve
+                    ? resolve(true)
+                    : reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+            };
+
+            const messageHandler = (event: MessageEvent) => {
+                if (event.origin === 'https://www.sandbox.paypal.com' || event.origin === 'https://www.paypal.com') {
```

Comment:
I think here you can also benefit from using a guard statement (to reduce the level of nesting). i.e.:

```ts
if (event.origin !== 'https://www.sandbox.paypal.com' && event.origin !== 'https://www.paypal.com') {
    return;
}
```